### ISSUE TITLE
Information about "in" function arguments missing

### ISSUE THREAD

**Issue by ssa3512** [dotnet/csharpstandard#264]
The `Argument Lists` section at https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#argument-lists references the `ref` and `out` parameter modifiers but is missing information on the newer `in` modifier which passes an item by reference but as read-only. The `in` modifier is described in the keywords section here: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/in-parameter-modifier

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e783650c-e063-e268-316e-564362cb1e37
* Version Independent ID: aa9f4996-cab4-0bed-6452-213342453785
* Content: [Expressions - C# language specification](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#argument-lists)
* Content Source: [spec/expressions.md](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md)
* Product: **dotnet-csharp**
* Technology: **csharp-spec**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **wiwagn**

---
[Associated WorkItem - 52212](https://dev.azure.com/msft-skilling/Content/_workitems/edit/52212)

**Comment by BillWagner** [comment]
I believe this will be addressed in #219

**Comment by KalleOlaviNiemitalo** [comment]
The C# standard should specify that the following does not throw ArrayTypeMismatchException at run time:

```csharp
static class Program {
    static void Main() {
        object[] array = new string[1];
        Sub(in array[0]); // Does not throw ArrayTypeMismatchException; the run-time type check is suppressed.
    }
    
    static void Sub(in object elt)
    {
        // Assigning elt = new object() here would violate type safety,
        // but the in modifier disallows this assignment.
    }
}
```

[SharpLab](https://sharplab.io/#v2:C4LglgNgPgAgjANgAQwEwrgdiQbwLABQSxGyMALEgLICGYAdgBQCUuhJHSA9gEYBWAUwDGwANoBdJDQBO0mgE8kAXiT0BAdwwAGUXHEBudp2IBlAK49GDKbIWit45oaIkAvkeIeUAZlIpK5pbWvIIiSAIQwMxe+C7E7gQJQA) shows that the IL code uses a `readonly.` prefix on the `ldelema` opcode to suppress the runtime check:

```
.class private auto ansi '<Module>'
{
} // end of class <Module>

.class private auto ansi abstract sealed beforefieldinit Program
    extends [System.Private.CoreLib]System.Object
{
    // Methods
    .method private hidebysig static 
        void Main () cil managed 
    {
        // Method begins at RVA 0x2050
        // Code size 22 (0x16)
        .maxstack 2
        .locals init (
            [0] object[]
        )

        IL_0000: ldc.i4.1
        IL_0001: newarr [System.Private.CoreLib]System.String
        IL_0006: stloc.0
        IL_0007: ldloc.0
        IL_0008: ldc.i4.0
        IL_0009: readonly.
        IL_000b: ldelema [System.Private.CoreLib]System.Object
        IL_0010: call void Program::Sub(object&)
        IL_0015: ret
    } // end of method Program::Main

    .method public hidebysig static 
        void Sub (
            [in] object& elt
        ) cil managed 
    {
        .param [1]
            .custom instance void [System.Private.CoreLib]System.Runtime.CompilerServices.IsReadOnlyAttribute::.ctor() = (
                01 00 00 00
            )
        // Method begins at RVA 0x2072
        // Code size 1 (0x1)
        .maxstack 8

        IL_0000: ret
    } // end of method Program::Sub

} // end of class Program
```

Alas, this code is not verifiable according to ECMA-335 6th Edition. §II.1.8.1.2.2 (Controlled-mutability managed pointers) says the result of `readonly. ldelema` "cannot be passed as a byref argument to a method." The verifier ignores the `[in]` ParamAttr in `[in] object& elt` because §II.15.4 (Defining methods) says "The CLI itself does not enforce the semantics of these bits".

**Comment by jskeet** [comment]
This should be fixed by #795.
